core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,304 @@
+import { Component, ElementRef, HostListener, ViewChild } from '@angular/core';
+import { AppConstants } from 'app.constants';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { UserService } from 'services/user.service';
+import { ContributionAndReviewStatsService } from '../services/contribution-and-review-stats.service';
+
+interface ContributionCounts {
+  language?: string;
+  submissions: number;
+  reviews: number;
+  corrections: number;
+}
+
+interface Badge {
+  contributionCount: number;
+  text: string;
+  language?: string;
+  isUnlocked: boolean;
+}
+
+@Component({
+  selector: 'contributor-badge',
+  templateUrl: './contributor-badge.component.html',
+  styleUrls: []
+})
+export class ContributorBadgeComponent {
+  @ViewChild('dropdown', {'static': false}) dropdownRef!: ElementRef;
+
+  @ViewChild(
+    'mobileDropdown', {'static': false}
+  ) mobileDropdownRef!: ElementRef;
+
+  @ViewChild(
+    'mobileBadgeTypeDropdown', {'static': false}
+  ) mobileBadgeTypeDropdownRef!: ElementRef;
+
+  totalTranslationStats: {[key: string]: ContributionCounts} = {};
+  translationBadges: {[key: string]: {[key: string]: Badge[]}} = {};
+  languages = [];
+  questionBadges: Badge[];
+  questionSubmissionBadges: Badge[];
+  questionReviewBadges: Badge[];
+  questionCorrectionBadges: Badge[];
+  totalQuestionStats: ContributionCounts = {
+    submissions: 0,
+    reviews: 0,
+    corrections: 0
+  };
+
+  dropdownShown = false;
+  mobileDropdownShown = false;
+  mobileBadgeTypeDropdownShown = false;
+  showMobileTranslationBadges = true;
```

Comment:
Can we make this an enum, e.g.

```suggestion
  enum MobileBadgeType {
    Question = "QUESTION",
    Translation = "TRANSLATION",
  }
```

It would make its usages clearer.

Reply:
Done